File: data/src/main/java/com/moebius/data/DocumentIndex.java
package com.moebius.data;

public enum DocumentIndex {

    TRADE_HISTORY("trade-stream-search"),
    TRADE_STAT("trade-stats-v2");

    private String index;

    DocumentIndex(String index){
        this.index = index;
    }

    @Override
    public String toString() {
        return index;
    }
}

File: backend/src/main/kotlin/com/moebius/backend/domain/commons/DocumentIndex.kt
package com.moebius.backend.domain.commons

import java.time.LocalDate

object DocumentIndex {
    val tradeStream: ElasticIndex = ElasticIndex("trade-stream",
            { name -> "$name-${LocalDate.now()}" },
{ name -> "$name-search" })
val tradeStats: ElasticIndex = ElasticIndex("trade-stats")

class ElasticIndex(
        private val name: String,
        private inline val indexDecorator: (String) -> String = { it -> it },
        private inline val searchAlias: (String) -> String = { it -> it }
) {
    fun searchIndex(): String {
        return searchAlias(name)
    }

    fun saveIndex(): String {
        return this.indexDecorator(name)
    }
}

}